## Importing necessary libraries

In [1]:
import pandas as pd
import numpy as np
import scipy.stats
from scipy.stats import norm
from scipy.stats import chi2_contingency

In [2]:
buyer_ratio_data = pd.read_csv('BuyerRatio.csv')       # Importing dataset
buyer_ratio_data

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

### Assume -  

**Null Hypothesis (Ho): All proportions are equal (male-female buyer rations are similar across regions does not vary and are not significantly related)**

**Alternate Hypothesis (Ha): Not all proportions are equal (male-female buyer rations are not similar across regions does vary and are significantly related)**

In [6]:
dimensional_data = np.array([[50,142,131,70],[435,1523,1356,750]])       # Make a contingency table
dimensional_data

array([[  50,  142,  131,   70],
       [ 435, 1523, 1356,  750]])

#### Chi2 contingency independence test

In [4]:
chi2_contingency(dimensional_data) # o/p is (Chi2 stats value, p_value, df, expected observations)

(1.595945538661058,
 0.6603094907091882,
 3,
 array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
        [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

### Comparing p-value with alpha(0.05)

### Inference

##### Since p_value 0.66 > 0.05

##### Null Hypothesis is accepted i.e. Independence of categorical variables.

##### Therefore, we can conclude that the male-female buyer rations are not similar across regions does not vary and are not significantly related.

**------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------**